### PR TITLE
Add prep functions man pages

### PR DESCRIPTION
**Pull request by heyrutvik**
Adds man pages for following prep functions (part of https://github.com/axboe/liburing/issues/777):

-  io_uring_prep_link_timeout
-  io_uring_prep_send_zc_fixed
-  io_uring_prep_sendmsg_zc
-  io_uring_prep_send_set_addr
-  io_uring_prep_getxattr
-  io_uring_prep_setxattr
-  io_uring_prep_fgetxattr
-  io_uring_prep_fsetxattr

### CODE REVIEW

**Comment by heyrutvik** [comment]
@axboe I tried to gather information reading tests and existing man pages. Though not so obvious information might be missing, like in what cases the request might fail because of not setting correct flags etc. Please take a look when you get a chance.
Also, the PR doesn't contain all the functions mentioned in #777. I'll try to add man pages for remaining functions in subsequent PRs.

File: test/link-timeout.c [hunk 1]
```diff
@@ -147,7 +147,7 @@ static int test_fail_two_link_timeouts(struct io_uring *ring)
 		case 4:
 			if (cqe->res != -ECANCELED && cqe->res != -EINVAL) {
 				fprintf(stderr, "Invalid link timeout got %d"
-					", wanted -ECACNCELED || -EINVAL\n", cqe->res);
+					", wanted -ECANCELED || -EINVAL\n", cqe->res);
```
Comment (ammarfaizi2) [inline test/link-timeout.c:150]: This is not a part of the manpage. Please split that into a separate patch.

File: test/link-timeout.c [hunk 2]
```diff
@@ -281,7 +281,7 @@ static int test_single_link_timeout_nop(struct io_uring *ring)
 			break;
 		case 2:
 			if (cqe->res != -ECANCELED) {
-				fprintf(stderr, "Link timeout got %d, wanted -ECACNCELED\n", cqe->res);
+				fprintf(stderr, "Link timeout got %d, wanted -ECANCELED\n", cqe->res);
```
Comment (ammarfaizi2) [inline test/link-timeout.c:284]: This is not a part of the manpage. Please split that into a separate patch.

**Review by ammarfaizi2 — CHANGES_REQUESTED** [review]
Please follow the pull request guidelines. The commit message must have a `Signed-off-by` tag.

**Comment by heyrutvik** [comment]
Thanks @ammarfaizi2 for reviewing. I'll update the PR/commit msg as per your remarks.

**Review by ammarfaizi2 — CHANGES_REQUESTED** [review]
A long man-page paragraph should be word-wrapped at 80 chars.

Some things should not be word-wrapped. They may be some kind of quoted
text, Link, oops reports, or whatever. Things that have a certain
specific format.

IOW, don't write a single-line long paragraph.

**Review by ammarfaizi2 — CHANGES_REQUESTED** [review]
Don't write "Adds man pages for following prep functions:" **as the
subject** because it will be used to generate a shortlog release.

Shortlog release is like this:
https://brick.kernel.dk/snaps/liburing-2.3.shortlog.txt

Saying "Adds man pages for following prep functions:" doesn't make
much sense with respect to the shortlog.

I suggest 1 commit per manpage. So it will look like this in the
shortlog release:

```
man: Add manpage for io_uring_prep_link_timeout()
man: Add manpage for io_uring_prep_send_zc_fixed()
man: Add manpage for io_uring_prep_sendmsg_zc()
man: Add manpage for io_uring_prep_send_set_addr()
...
...
...
```

**Review by ammarfaizi2 — CHANGES_REQUESTED** [review]
Please align the argument list properly like the other manpages.

For example, you wrote this (pay attention to second argument to the
last argument):

```
.BI "void io_uring_prep_getxattr(struct io_uring_sqe *" sqe ","
.BI "                        const char *" name ","
.BI "                        char *" value ","
.BI "                        const char *" path ","
.BI "                        unsigned int " len ");"
.PP
.BI "void io_uring_prep_fgetxattr(struct io_uring_sqe *" sqe ","
.BI "                        int" fd ","
.BI "                        const char *" name ","
.BI "                        char *" value ","
.BI "                        unsigned int " len ");"
```

It would be nice to have them aligned properly like this (add more spaces so
they are aligned):

```
.BI "void io_uring_prep_getxattr(struct io_uring_sqe *" sqe ","
.BI "                            const char *" name ","
.BI "                            char *" value ","
.BI "                            const char *" path ","
.BI "                            unsigned int " len ");"
.PP
.BI "void io_uring_prep_fgetxattr(struct io_uring_sqe *" sqe ","
.BI "                             int" fd ","
.BI "                             const char *" name ","
.BI "                             char *" value ","
.BI "                             unsigned int " len ");"
```

**Comment by heyrutvik** [comment]
@ammarfaizi2  Apologies for these silly mistakes. I've updated the PR.

File: man/io_uring_prep_setxattr.3 [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+argument specifies the size (in bytes) of
+.I value.
+
+.BR io_uring_prep_fsetxattr(3)
+is identical to 
+.BR io_uring_prep_setxattr(3),
+only the extended attribute is set on the open file referred to by
+.I fd
+in place of
+.I path.
```
Comment (ammarfaizi2) [inline man/io_uring_prep_setxattr.3:53]: This dot after "`path`" should not be underlined, it should be a roman
font.

That means, don't write this:
```
.I path.
```

But write this:
```
.IR path .
```

Just like what we have here (this is an example where you have a dot after mentioning a variable):
https://github.com/axboe/liburing/blob/b82c5b77cd34edbfdb526b83ac1bad1935647299/man/io_uring_prep_recv.3#L38

Side note:
`.I` typesets in italics (but terminals show that as underline), and `.R`
in what troff calls the roman font, and the rest of us call the normal
font.

You can combine these as `.IR` (the first argument is italic, the second is
roman).

File: man/io_uring_prep_setxattr.3 [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+of the extended attribute identified by
+.I name
+and associated with the given
+.I path
+in the filesystem with modifier flags
+.I flags.
+The
+.I len
+argument specifies the size (in bytes) of
+.I value.
```
Comment (ammarfaizi2) [inline man/io_uring_prep_setxattr.3:45]: The same font issue.

File: man/io_uring_prep_link_timeout.3 [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+.BI "                                unsigned " flags ");"
+.fi
+.SH DESCRIPTION
+.PP
+The 
+.BR io_uring_prep_link_timeout(3)
+function prepares a timeout request for linked sqes. The submission queue entry 
+.I sqe
+is setup a timeout specified by
+.I ts.
```
Comment (ammarfaizi2) [inline man/io_uring_prep_link_timeout.3:23]: The same font issue.

File: man/io_uring_prep_setxattr.3 [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+queue entry 
+.I sqe
+is setup to set the
+.I value
+of the extended attribute identified by
+.I name
+and associated with the given
+.I path
+in the filesystem with modifier flags
+.I flags.
```
Comment (ammarfaizi2) [inline man/io_uring_prep_setxattr.3:41]: The same font issue.

**Review by ammarfaizi2 — CHANGES_REQUESTED** [review]
> @ammarfaizi2 Apologies for these silly mistakes. I've updated the PR.

It's fine. Thank you for doing this!

File: man/io_uring_prep_setxattr.3 (hunk 3 above)
Comment (heyrutvik) [reply man/io_uring_prep_setxattr.3]: I see. Thanks for explaining it. I was just looking at the terminal whether it was formatted properly or not.

File: man/io_uring_prep_link_timeout.3 [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+.PP
+The 
+.BR io_uring_prep_link_timeout (3)
+function prepares a timeout request for linked sqes. The submission queue entry 
+.I sqe
+is setup a timeout specified by
+.IR ts .
+The flags argument holds modifier
+.I flags
+for the request.
```
Comment (ammarfaizi2) [inline man/io_uring_prep_link_timeout.3:26]: > The flags argument holds modifier
> .I flags
> for the request.

Since there is no corresponding syscall entry manpage for this, explain
what the `flags` modifier means here. And what possible values for it?

Comment (ammarfaizi2) [inline man/io_uring_prep_link_timeout.3:26]: Add the `struct __kernel_timespec` definition in the DESCRIPTION section.

**Review by ammarfaizi2 — CHANGES_REQUESTED** [review]

File: man/io_uring_prep_getxattr.3 [hunk 8]
```diff
@@ -0,0 +1,61 @@
+.\" Copyright (C) 2023 Rutvik Patel <heyrutvik@gmail.com>
+.\"
+.\" SPDX-License-Identifier: LGPL-2.0-or-later
+.\"
+.TH io_uring_prep_getxattr 3 "January 23, 2023" "liburing-2.3" "liburing Manual"
```
Comment (ammarfaizi2) [inline man/io_uring_prep_getxattr.3:5]: All of these should go to liburing-2.4 or later.

liburing-2.3 is not being redone.

**Comment by axboe** [comment]
@heyrutvik Thanks a lot for doing this work! Do you have time to make the last few edits so we can get this included?

**Comment by heyrutvik** [comment]
@axboe Just updated. I missed @ammarfaizi2's last remarks.

**Comment by axboe** [comment]
Thanks! Now pulled.

**Comment by axboe** [comment]
If you're up for it, there's io_uring_prep_msg_ring_fd() as well now, as I forgot to add this helper initially...

**Comment by heyrutvik** [comment]
@axboe I'll add that to the original issue. Thanks.
